### DISCUSSION TITLE
Bringing MetaMask to Akash via The Cosmos Snap

### DISCUSSION THREAD

**Discussion by JohnDiBernardi** [akash-network/community#247] [Governance Proposals]
**Goal:**
This is a project that will give over 30M active monthly Metamask users seamless access to the Akash ecosystem. The Cosmos Snap for MetaMask will also reduce the barriers to entry for collaboration between Akash-native and EVM-native projects. With increased accessibility into both the largest DeFi ecosystem and NFT marketplaces, this will open the door for further dApp exposure and uniquely position Cosmos alongside Ethereum as a driving force for general adoption in the blockchain industry. This proposal is focused on properly integrating Akash into the Cosmos Snap wallet that will live on the metamask.mysticlabs.xyz domain.

**Grant & Uses:**
Amount: $50,000 USD worth of AKT tokens sold over time (in correspondence with the Akash team)

**Use:**
Development of Akash-specific and highly requested features, which include (but are not limited to): ensuring all Akash-native assets are natively supported in the Cosmos Snap, support for in-wallet notifications of chain-related activity, multi-sig support in-wallet, and alternate frontend hosted on an Akash web server. Also, any funds left over from development funding will be set aside for payment of the web server over time to ensure its longevity. 

**Deliverable for Akash:** A proper integration into the Cosmos Snap for Akash to give MetaMask users the ability to access the Akash Marketplace, in turn enabling the buying and selling of decentralized cloud computing infrastructure to Ethereum-based users, as well as increase user exposure to testing advanced AI models once the rollout of Akash AI is completed. Support for in-wallet notifications will provide a much greater user experience for those who use the Cosmos Snap, since notifications such as low balance warnings on existing web servers, instance renewals, and more are extremely important to Akash users. Also, an alternate wallet UI hosted on Akash will provide more exposure to the capabilities of Akash when using (Powered by Akash banner located somewhere on page). Multi-sig support is an added bonus that not only increases overall wallet capabilities for Akash users, but for the entire Cosmos as well due to the lack of support for a well-established Cosmos-native multi-sig option. 

**Milestones/Proposed Payment Schedule:**
40% of ask - paid up front to provide enough runway for initial development costs

Remaining funds (60%) to be disbursed by selected members of the Akash team broken down as follows:
20% of ask - paid upon completion of successful integration of notification support
20% of ask - paid upon deployment of the alternate UI on an Akash web server
20% of ask - paid upon completion of in-wallet multi-sig support

**Benefits of This Proposal for Akash:**

1. A direct connection to the 30M+ monthly active users currently using MetaMask
2. In-wallet notification support for functions such as low-balance indicators
3. Multi-sig support directly in-wallet
4. An alternate wallet UI hosted on an Akash-based server
5. Access to Akash's decentralized cloud-computing for Ethereum-based users
6. Enhance the security of the Akash chain by extending the opportunity for staking AKT tokens to the crypto industry’s most capitalized user base.
7. Potentially increase Akash Marketplace sales volume due to increased exposure to Ethereum-based users
8. Further strengthen the link between Akash, Cosmos, and Ethereum

For more information on Mystic Labs, head on over to https://www.mysticlabs.xyz/.

You can view the current GitHub repo for the Cosmos Snap here: https://github.com/cosmos/snap

Feel free to tag me below in the comments or tag @Mystic_Labs_ on Twitter for more insight into what Mystic Labs does and how we are bringing value to the Akash & Cosmos Ecosystems.

**Comment by gosuri** [comment]
This looks amazing! Looking forward to discussing this prop at the steering committee meeting.

**Reply by JohnDiBernardi** [reply]
Appreciate the support❤️ Can't wait to get this on-chain!

**Comment by RobAlGhul** [comment]
Grant & Uses:
Amount: $50,000 USD worth of FLIX tokens and $50,000 worth of AKT tokens sold over time (in correspondence with OmniFlix & Akash teams)

You cannot ask for $50k USD from the OmniFlix community with only $218K of liquidity available. Please revise and do your research before posting.

**Reply by JohnDiBernardi** [reply]
Understand the concern, we have been coordinating with the OmniFlix core team and will sell tokens on an agreed-upon schedule in order to prevent any liquidity constraints within the ecosystem (assuming the proposal is passed)

While Co-Promoted, this proposal is technically independent of the OmniFlix prop, so if the OmniFlix community decides it's not in their best interest then so be it, we would proceed with just the Akash portion (though it was in the best interest of both teams to have this proposal Co-Sponsored because of the similarities in the implementation itself as well as their positive working relationship)

**Reply by RobAlGhul** [reply]
" to prevent any liquidity constraints" - sirrr, there is already a liquidity constraint. You are asking for 25% of the liquidity from the FLIX community. https://info.osmosis.zone/token/FLIX

**Comment by PolkachuIntern** [comment]
I like the idea and the project. It will have a huge impact in the ecosystem. However, the proposal can be improved with the following:

1. Lay out the total budget. It does not need to have precise line-items, but the community should be aware of the total ballpark budget.
2. lay out the grants already received and grants that the team is still planning to apply. Ideally, this should match or be slightly above the budget. 
3. Make the grant milestone-based. For example, a portion for initial development, a portion for audit completion, a portion for testnet launch and a port for mainnet launch. We are even have a separate signal proposal for on-going monthly maintenance cost too.

**Comment by anilmurty** [comment]
This is well written and answers all my questions. Thanks for writing it. Looking forward to the live discussion later this week.

**Comment by zJuuu** [comment]
Thank you for your short presentation in the Steering Comittee Call.
I really like the idea of implementing Akash in to MM.
In the call you wrote in the chat, that it will be possible to "swap directly from any ETH-native asset to any Cosmos-native asset directly in wallet". 
How will this be implemented?

**Reply by JohnDiBernardi** [reply]
Thanks for the response here, was a pleasure presenting to you all

There will be an in-wallet swaps interface that provides EVM-Cosmos and Cosmos-Cosmos swaps using your existing MM assets. This removes another friction point currently experienced by EVM users looking to get into the Cosmos ecosystem

**Reply by JohnDiBernardi** [reply]
This will be available for all to use on Day 1

**Reply by zJuuu** [reply]
That's great! Will the alternate wallet UI be open source?

**Reply by JohnDiBernardi** [reply]
Yup!

**Comment by dylanschultzie** [comment]
Hey @JohnDiBernardi, thanks for your proposal. I may have missed it in the discussion, but is this proposal to cover Akash support permanently? Or over a year? 2 years?

**Reply by JohnDiBernardi** [reply]
Permanent support, this doesn’t guarantee permanent RPC support as those are separate (contract will be public soon for that) but Akash itself will be supported as long as cosmos is (which would be the lifetime of the Snap, since it wouldn’t be a cosmos snap without it)

**Reply by dylanschultzie** [reply]
Wait - so this requires a separate proposal for RPCs?

**Reply by JohnDiBernardi** [reply]
Nope, RPCs are all covered now

**Reply by JohnDiBernardi** [reply]
Funded until breakeven is reached by AADAO

**Reply by dylanschultzie** [reply]
Roger, thank you!

**Comment by Lucas-Kohorst** [comment]
hey this is super cool especially with metamask's beta of snaps today (https://snaps.metamask.io) also seems like there are a few cosmos based snaps out (https://snaps.metamask.io./snap/npm/cosmsnap/snap/, https://snaps.metamask.io./snap/npm/vegaprotocol/snap/) 

Curious on the status of this proposal

**Comment by JohnDiBernardi** [comment]
Hey everybody, wanted to make an update here:

We have successfully completed and delivered all items related to this proposal, and the final integration is slated to be merged to main on Akash console soon!

Here's a video showing how to create a deployment: [Open Lease/Use Akash with MetaMask](https://www.loom.com/share/cbb644d9ca614bc7a0bed1772f5e2ec1?sid=33c65065-bb04-4432-937d-f339ddff6dfc)

Here's a video showing how to view your leases and notifications directly in the extension: [View Leases/Notifications in Extension](https://www.loom.com/share/ab2d233fbfc3457bbeca60217fb4951a?sid=56e614ca-23d7-48e5-9833-c8241cf2ee51)


Multisig links:
How to self host: https://github.com/cosmos/snap/blob/multisig/SELF_HOST.md 
Live alternate UI for Akash: https://hhaugnap79cjv83ceto55qn4d0.ingress.d3akash.cloud/ 
Video of the multisig: https://www.loom.com/share/ef49259947134b3b91b1215b74c1d3ec?sid=3ae3ff36-b164-4f70-9221-ee786380555e 


You can now manage multisigs for any Cosmos chain in one place, all hosted on Akash and with Akash used by default. You have the ability to IBC transfer, send, track transactions, one click swaps, and even add multisig functionality within other dApps (using the Cosmos Metamask Provider called [snapper](https://www.npmjs.com/package/@cosmsnap/snapper) + window.cosmos.createMultisig).

You can also self-host the multisig (using mysticlabss/multisig:v0.1.0 ) on Akash (including the dependency Appwrite instance) if you are a business that prefers to do such and don't want to use our deployment.

Akash console github link(s): https://github.com/akash-network/console/issues/62 
https://github.com/akash-network/console/pull/291 
https://github.com/akash-network/console/pull/334 



There will be another proposal made soon to distribute the rest of the funds from this proposal. This will be brought up during today's (Aug 29th) steering committee meeting.

Excited to see this in action and give a better experience for Akash users!

**Reply by anilmurty** [reply]
video demos look great - can't wait to try it!

**Comment by JohnDiBernardi** [comment]
Hey guys quick update, proposal is now live: https://www.mintscan.io/akash/proposals/266

Please get your votes in!

**Reply by piber-dev** [reply]
hi @JohnDiBernardi after seeing so many NO-votes, i took a more in-depth look at the proposal and i'm guessing the reason to why there are so many NO-votes is that the ask for "60% of $50k in AKT tokens" is tied to the value of AKT tokens when you made your first proposal. this is an current ask of $75k which added to the ~20k requested last time is >$100k USD at the time of requesting funds. while i did not attend last months steering committee meeting and there not being any notes available yet, it would make sense to have the ask be $30k of AKT based off what you wrote in this discussion. is there a reason to why future asks are tied to the number in tokens and not the number in USD when you specifically asked for "$50k worth of tokens" not "X amount of tokens"?

**Reply by PolkachuIntern** [reply]
I confirm that this was why Polkachu voted no

**Comment by JohnDiBernardi** [comment]
Hey everyone, we have decided to repost the prop to reflect the updated amount as recommended by the community.

Please vote here! https://www.mintscan.io/akash/proposals/271

**Comment by brewsterdrinkwater** [comment]
This proposal passed on chain: https://www.mintscan.io/akash/proposals/271 

The feature is live on Akash Console.
